### PR TITLE
ci: add nightly build workflow

### PR DESCRIPTION
**Pull request by burkedavison**
Issue #1274

### CODE REVIEW

**Comment by burkedavison** [comment]
Before replacing the pull_request trigger with a nightly cron trigger:
image

**Review by meltsufin — APPROVED** [review]

File: gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java [hunk 1]
```diff
@@ -68,6 +68,7 @@
 
 @RunWith(JUnit4.class)
 public class HttpJsonDirectServerStreamingCallableTest {
+
```
Comment (suztomo) [inline gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java:71]: remove

File: gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java [hunk 2]
```diff
@@ -323,6 +328,7 @@ public void testBlockingServerStreaming() {
   }
 
   static class MoneyObserver extends StateCheckingResponseObserver<Money> {
+
```
Comment (suztomo) [inline gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java:331]: remove

**Review by suztomo — APPROVED** [review]
Let's run it to see any failures.

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+defaults:
+  run:
+    shell: bash
+jobs:
+  nightly:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macOS-latest, ubuntu-latest, windows-latest ]
```
Comment (suztomo) [inline .github/workflows/nightly.yaml:14]: Shall we avoid "-latest"? In my experience, it suddenly stops working when GitHub infrastructure changes what is latest.

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

File: gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java (hunk 1 above)
Comment (burkedavison) [reply gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonDirectServerStreamingCallableTest.java:71]: > Let's run it to see any failures.

https://github.com/googleapis/gapic-generator-java/actions/runs/4010427729/jobs/6887076787

This is the commit before the swap to scheduling via cron.

**Comment by burkedavison** [comment]
> Let's run it to see any failures.

https://github.com/googleapis/gapic-generator-java/actions/runs/4010427729/jobs/6887076787

This is the commit before the swap to scheduling via cron.

File: .github/workflows/nightly.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+name: Nightly
+on:
+  pull_request:
```
Comment (suztomo) [inline .github/workflows/nightly.yaml:3]: It runs on every pull requests?

Add "workflow_dispatch" so that we can invoke the job manually.

**Review by suztomo — APPROVED** [review]

File: .github/workflows/nightly.yaml (hunk 4 above)
Comment (burkedavison) [reply .github/workflows/nightly.yaml:3]: But also:
> To trigger the workflow_dispatch event, your workflow must be in the default branch.

So `pull_request` has been useful to develop this in the PR, but I'm now replacing with `workflow-dispatch`.

File: .github/workflows/nightly.yaml (hunk 3 above)
Comment (burkedavison) [reply .github/workflows/nightly.yaml]: Done

**Pull request merged**